src/wallet/view_seed_flow.c

```diff
@@ -0,0 +1,150 @@
+#include "ui_list.h"
+#include "ui_multi_instruction.h"
+#include "ui_state_machine.h"
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+/**
+ * @brief The function handles transitions of view seed flow for the passed
+ * wallet id.
+ *
+ * @param wallet_id A pointer to a uint8_t array representing the wallet ID.
+ *
+ * @return true if view seed is successful, else false.
+ */
+static bool view_seed_handler(const uint8_t *wallet_id);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+static bool view_seed_handler(const uint8_t *wallet_id) {
+  CONFIDENTIAL char mnemonics[MAX_NUMBER_OF_MNEMONIC_WORDS]
+                             [MAX_MNEMONIC_WORD_LENGTH] = {0};
+  CONFIDENTIAL uint8_t no_of_mnemonics = 0;
+  bool result = false;
+
+  do {
+    multi_instruction_init(
+        ui_text_view_seed_messages, 3, DELAY_LONG_STRING, true);
+    if (0 != get_state_on_confirm_scr(0, 1, 2)) {
+      break;
+    }
+
+    no_of_mnemonics = reconstruct_mnemonics_flow(wallet_id, mnemonics);
+
+    if (12 != no_of_mnemonics && 18 != no_of_mnemonics &&
+        24 != no_of_mnemonics) {
+      break;
+    }
+
+    set_theme(LIGHT);
+    list_init(mnemonics, no_of_mnemonics, ui_text_word_hash, true);
+    if (0 == get_state_on_confirm_scr(0, 1, 2)) {
+      result = true;
+      break;
+    }
+  } while (0);
+
+  reset_theme();
+  memzero(mnemonics, sizeof(mnemonics));
+  no_of_mnemonics = 0;
+  return result;
+}
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+void view_seed_flow(uint8_t *wallet_id) {
+  ASSERT(NULL != wallet_id);
+
+  if (false == view_seed_handler(wallet_id)) {
+    mark_core_error_screen(ui_text_something_went_wrong);
```

Comment:
Instead of adding error here.. we should add this error screen in reconstruct_wallet_flow handler.. in case it reaches state `COMPLETED_WITH_ERRORS`.. so that this handling is done in every flow which calls the reconstruct_wallet_flow

Reply:
I'll add a todo, and we can take it up as a follow up change.

Reply:
updated